roles/openshift_health_checker/openshift_checks/docker_storage.py

```diff
@@ -0,0 +1,113 @@
+# pylint: disable=missing-docstring
+import json
+
+from openshift_checks import OpenShiftCheck, get_var
+
+
+class DockerStorage(OpenShiftCheck):
+    """Check Docker storage sanity.
+
+    This check ensures that thinpool usage is not
+    close to 100% for a containerized installation.
+    """
+
+    name = "docker_storage"
+    tags = ["preflight"]
+
+    max_thinpool_data_usage_percent = 90.0
+    max_thinpool_metadata_usage_percent = 90.0
+
+    @classmethod
+    def is_active(cls, task_vars):
+        """Skip non-containerized installations."""
+        is_containerized = get_var(task_vars, "openshift", "common", "is_containerized")
+        return super(DockerStorage, cls).is_active(task_vars) and is_containerized
+
+    def run(self, tmp, task_vars):
+        self.max_thinpool_data_usage_percent = get_var(task_vars,
+                                                       "max_thinpool_data_usage_percent",
+                                                       default=self.max_thinpool_data_usage_percent)
+        self.max_thinpool_metadata_usage_percent = get_var(task_vars, "max_thinpool_metadata_usage_percent",
+                                                           default=self.max_thinpool_metadata_usage_percent)
+
+        failed, msg = self.check_thinpool_usage(task_vars)
+        if failed:
+            return {"failed": True, "msg": msg}
+
+        return {"changed": False}
+
+    def check_thinpool_usage(self, task_vars):
+        no_data_msg = "no thinpool usage data returned by the host."
+
+        failed, data, msg = self.get_lvs_data(task_vars)
+        if failed:
+            if not msg:
+                msg = no_data_msg
+            return failed, msg
+
+        lv = self.extract_thinpool_obj(data)
+        if not lv:
+            return True, no_data_msg
+
+        failed, data_percent = self.get_thinpool_data_usage(lv)
+        if failed:
+            return failed, no_data_msg
+
+        failed, metadata_percent = self.get_thinpool_metadata_usage(lv)
+        if failed:
+            return failed, no_data_msg
+
+        if data_percent > float(self.max_thinpool_data_usage_percent):
```

Comment:
When we read `max_thinpool_data_usage_percent` from `task_vars`, as the tests show, it might be a string... and we can't always convert a string to float (not tested).
We could be a bit more careful and handle `TypeError`.

I'd suggest doing that early on when we read from `task_vars`, not here.